src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -184,8 +196,16 @@ const useCoreChatSystemHook = () => {
           newWebsocketData,
           channelType[websocketData?.channel_type]
         );
-        const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
-        await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+
+        queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.CoreChatSystem_GeneralSystemMessage,
+          id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+          priority: QueueJobPriority.HIGH,
+          task: async () => {
+            const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+            await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+          }
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, database operations are now being added to a priority queue. Ensure that the queue is well-managed and that high-priority tasks do not starve lower-priority tasks.

```diff
- const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
- await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+ queue.addPriorityJob({
+   operationLabel: DatabaseOperationLabel.CoreChatSystem_GeneralSystemMessage,
+   id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+   priority: QueueJobPriority.HIGH,
+   task: async () => {
+     const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+     await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+   }
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->